doc/test-machine.md

```diff
@@ -135,11 +146,43 @@ better that you write this macro yourself so that you can tailor it to your own
 requirements.
 
 ```clojure
-(defn with-test-machine [f {:keys [transport}]}]
-  (fix/with-fixtures [(fix/topic-fixture kafka-config input-topics)
-                      (fix/topic-fixture kafka-config output-topics)
-                      (fix/service-ready {:http-url "http://localhost:8082"
-                                          :http-timeout 5000})]
-    (with-open [machine (test-machine {:transport transport})]
+(ns my.app-test
+  (:require
+    [my.app :as app]
+    [jackdaw.serdes :refer [string-serde]]
+    [jackdaw.test :refer [test-machine]]
+    [jackdaw.test.fixtures :refer [with-fixtures topic-fixture service-ready?]])
+  (:import
+    (org.apache.kafka.streams TopologyTestDriver)))
+
+(def kafka-config
+  {"bootstrap.servers" "localhost:9092"
+   "group.id" "my-app"})
+
+(def input-topic-config
+  {:foo {:topic-name "foo"
+         :key-serde (string-serde)
+         :value-serde (string-serde)
+         :partition-count 1
+         :replication-factor 1})
+
+(def output-topic-config
+  {:foo {:topic-name "bar"
+         :key-serde (string-serde)
+         :value-serde (string-serde)
+         :partition-count 1
+         :replication-factor 1})
+
+(defn with-test-machine
+  "Creates a test-machine using the supplied `transport` and then
+   passes it to the supplied `f`."
+  [f transport]
+  (with-fixtures [(topic-fixture kafka-config input-topic-config)
+                  (topic-fixture kafka-config output-topic-config)
+                  (service-ready? {:http-url "http://localhost:8082"
+                                   :timeout 5000})]
+    (with-open [machine (test-machine transport)]
       (f machine))))
 ```
+
+The `topic-fixture` function creates all the topics named in the supplied `topic-config` before running tests. You have to import `TopologyTestDriver` for `test-machine` to work, which requires you to bring in the `org.apache.kafka/kafka-streams-test-utils` library as a dependency, using a version within `2.0.0` - `2.3.0`. The `topic-fixture` expects the `topic-config` to contain `:partition-count` and `:replication-factor` to be present, besides the `topic-name` and key-value serdes.
```

Comment:
Why not above than `2.3.0`?

Reply:
Thanks @gphilipp  - Currently, when bringing in versions of `kafka-streams-test-utils` higher than `2.3.0`,  it throws errors. For example, for `2.4.0`:
```
Syntax error (ClassNotFoundException) compiling new at (jackdaw/test.clj:250:5).
org.apache.kafka.streams.internals.QuietStreamsConfig
```
I assume we need to bring Jackdaw up to date with the newest versions from Apache Kafka's libraries.

I tried all version numbers from `2.4.0, 2.5.0` ... up to the latest `2.8.0` - they all throw errors in `jackdaw/test.clj`.

I tested this in a dummy example app, you can find it here: 
https://github.com/FundingCircle/jackdaw-examples

Reply:
Ok then, let's file a separate issue for that then 👍 .

Reply:
OK, opened an issue here: https://github.com/FundingCircle/jackdaw/issues/290